### PR TITLE
Adding code of conduct and development documentation

### PR DESCRIPTION
**Pull request by gvegayon**
This pull request includes significant updates to the project's documentation, focusing on the Code of Conduct and development guidelines. The most important changes include the addition of a comprehensive Code of Conduct, updates to the development documentation, and modifications to the README file to reference these new documents.

### Documentation Updates:

* [`CODE_OF_CONDUCT.md`](diffhunk://#diff-ffdbe3a1e7ee93cacfc080b6c635ccf3a8f6b0f00f2fb884f78c6b5f9dac8fd2R1-R126): Added a detailed Contributor Covenant Code of Conduct, outlining community standards, enforcement responsibilities, and guidelines for addressing unacceptable behavior.

* [`DEVELOPMENT.md`](diffhunk://#diff-fc2a310fcfedfefb0046def697f932def80cbb1e78c689bc0af3c8eab3e33eceR1-R17): Introduced new development documentation, including guidelines for contributing, instructions for installing pre-commit hooks, and notes for building the container image.

### README Modifications:

* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L28-R30): Updated to include a reference to the new Code of Conduct and a link to the development guidelines. Removed the section on building the container image, which is now covered in `DEVELOPMENT.md`.

### CODE REVIEW

**Comment by gvegayon** [comment]
@apulsipher, some links in the `DEVELOPMEN.md` file are broken but should be OK once this is merged. If not, we can create an issue to address that after we merge this.

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+This Code of Conduct applies within all community spaces, and also applies when
+an individual is officially representing the community in public spaces.
+Examples of representing our community include using an official e-mail address,
+posting via an official social media account, or acting as an appointed
+representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the community leaders responsible for enforcement at @pulsipher.
```
Comment (apulsipher) [inline CODE_OF_CONDUCT.md:62]: should be @apulsipher

File: README.md [hunk 2]
```diff
@@ -25,9 +25,6 @@ It's worth also taking a look at:
...
 - [Epinowcast](https://www.epinowcast.org) and the Epinowcast [community forums](https://community.epinowcast.org)
 
-## Building Container Image
-Use the Dockerfile in [`.devcontainer/`] to build the `epiworld-forecasts` image.
-The image includes [`rocker/tidyverse`](https://rocker-project.org/images/versioned/rstudio.html) and [`epiworldR`](https://github.com/UofUEpiBio/epiworldR).
-It is stored on the Github Container Registry (`ghcr.io/`).
+## Code of Conduct
 
-**Note for macOS users (M-series chip):** Pulling and building the `rocker/tidyverse` image may require specifying the platform with `--platform=linux/amd64`
+Please note that the epiworld-forecasts project is released with a [Contributor Code of Conduct](https://contributor-covenant.org/version/2/1/CODE_OF_CONDUCT.html). By contributing to this project, you agree to abide by its terms. More information about how to contribute to the project can be found under [`DEVELOPMENT.md`](DEVELOPMENT.md).
```
Comment (apulsipher) [inline README.md:30]: Should this link to our code of conduct instead of this website?

**Review by apulsipher — COMMENTED** [review]
Two things to fix, but otherwise looks great.

**Pull request merged**